framework/src/loops/ComputeMortarFunctor.C

```diff
@@ -267,6 +272,9 @@ ComputeMortarFunctor::operator()()
         for (auto * const mc : _mortar_constraints)
         {
           mc->setNormals(mc->interpolateNormals() ? normals : nodal_normals);
+          mc->setSecondaryIpToLowerdMap(secondary_ip_to_lowerd_elemen_nodal_map);
+          mc->setPrimaryIpToLowerdMap(primary_ip_to_lowerd_elemen_nodal_map);
+          mc->setNodalTangents(nodal_tangents);
```

Comment:
This is my design so I'm not going to request that you change it, but seeing all these setters and getters used in this loop class makes me think that we should just have a reference to `AutomaticMortarGeneration` in the mortar constraint class. Then we could cut out the loop middle-man. Anyway that could be a follow-on cleanup

Reply:
I'll take a stab at this in a new commit.